app/controllers/sessions_controller.rb

```diff
@@ -1,7 +1,7 @@
 class SessionsController < Devise::SessionsController
-  skip_before_filter :verify_authenticity_token, only: [:destroy], if: :json_request?
-  after_filter :set_csrf_headers, only: [:create, :destroy]
-  after_filter :set_csrf_headers, only: :new, if: :json_request?
+  skip_before_action :verify_authenticity_token, if: -> { json_request? && action_name == 'destroy' }
```

Comment:
OOI - did the `only: [:destroy]` option change as the API for these callbacks has changed? I.e. we inlined the only option as outlined in https://api.rubyonrails.org/v5.1.0/classes/AbstractController/Callbacks/ClassMethods.html#method-i-_normalize_callback_options

Reply:
I believe `only: [:destroy]` is still valid syntax. Though per https://www.rubydoc.info/gems/rubocop-rails/RuboCop/Cop/Rails/IgnoredSkipActionFilterOption, `if` option will be ignored.

Reply:
Actually do we want `if` to be ignored? if so, we can ignore hound sniff and I can revert it back.

Reply:
no my reading of the original syntax is we want this this action to be skipped only if the actions is a destroy action and only if it's a json request so we want both checks to run. 

What this skip action is doing is avoiding rails in built CSRF checks / errors for json API calls on logout actions, https://github.com/zooniverse/panoptes/pull/1424